code/modules/mob/living/carbon/need_procs.dm

```diff
@@ -0,0 +1,129 @@
+/mob/living/carbon/proc/print_happiness()
+	var/msg = "<span class='info'>*---------*\n<EM>Your current mood</EM>\n"
+	for(var/i in events)
+		var/datum/happiness_event/event = events[i]
+		msg += event.description
+	to_chat(src, msg)
+
+/mob/living/carbon/proc/update_happiness()
+	var/old_happiness = happiness
+	happiness = 0
+	for(var/i in events)
+		var/datum/happiness_event/event = events[i]
+		happiness += event.happiness
+
+	if(happiness > old_happiness)
+		to_chat(src, "<span class='green'>You mood improves.</span>")
+	else if(happiness < old_happiness)
+		to_chat(src, "<span class='warning'>You mood gets worse.</span>")
+	switch(happiness)
+		if(-5000000 to MOOD_LEVEL_SAD4)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood1"
+		if(MOOD_LEVEL_SAD4 to MOOD_LEVEL_SAD3)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood2"
+		if(MOOD_LEVEL_SAD3 to MOOD_LEVEL_SAD2)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood3"
+		if(MOOD_LEVEL_SAD2 to MOOD_LEVEL_SAD1)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood4"
+		if(MOOD_LEVEL_SAD1 to MOOD_LEVEL_HAPPY1)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood5"
+		if(MOOD_LEVEL_HAPPY1 to MOOD_LEVEL_HAPPY2)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood6"
+		if(MOOD_LEVEL_HAPPY2 to MOOD_LEVEL_HAPPY3)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood7"
+		if(MOOD_LEVEL_HAPPY3 to MOOD_LEVEL_HAPPY4)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood8"
+		if(MOOD_LEVEL_HAPPY4 to INFINITY)
+			if(client && hud_used)
+				hud_used.happiness.icon_state = "mood9"
```

Comment:
I've told you this about icon states before: `icon_state = "[initial(icon_state)][var]"`

Reply:
what would that change in this case? I'm not really using a var as number. Just curious by the way. not trying to say my method is better

Reply:
initial() is better in general so subtypes can just change the initial icon_state to have a different look. the number at the end, if you can make it scale right with mood levels shortens this entire list a ton. Based on a quick glance at the mood level defines you could probably do something like `icon_state = "[initial(icon_state)][round(happiness, 5)/5]"`

Reply:
I've changed the defines so that probably won't work anymore. they'll also continue to keep changing due to the huge problems with balance we'll have whenever we add or remove moodlets.